src/hooks/useActionsCount.ts

```diff
@@ -0,0 +1,61 @@
+import { startOfDay, subDays } from 'date-fns';
+
+import { useGetTotalColonyActionsQuery } from '~gql';
+import { getDomainDatabaseId } from '~utils/databaseId';
+
+import { getBaseSearchActionsFilterVariable } from './useActivityFeed/helpers';
+import useColonyContext from './useColonyContext';
+
+const getThirtyDaysAgoIso = () => {
+  const thirtyDaysAgo = subDays(new Date(), 30);
+  const midnightThirtyDaysAgo = startOfDay(thirtyDaysAgo);
+  return midnightThirtyDaysAgo.toISOString();
+};
+
+interface UseActionsCountParams {
+  domainId?: number;
+  onlyRecent?: boolean;
+}
+
+/**
+ * Hook returning the total number of actions matching the given params
+ */
+const useActionsCount = (params?: UseActionsCountParams) => {
+  const { domainId, onlyRecent } = params ?? {};
+  const { colony } = useColonyContext();
+  const { colonyAddress = '' } = colony ?? {};
+
+  const { data: totalActionData } = useGetTotalColonyActionsQuery({
+    variables: {
+      filter: {
+        ...getBaseSearchActionsFilterVariable(colonyAddress),
+        or: domainId
+          ? [
+              {
+                fromDomainId: {
+                  eq: getDomainDatabaseId(colonyAddress, domainId),
+                },
+              },
+              {
+                motionDomainId: { eq: domainId },
+              },
+            ]
+          : undefined,
+        createdAt: onlyRecent
+          ? {
+              gte: getThirtyDaysAgoIso(),
+            }
+          : undefined,
+      },
+    },
+    skip: !colony,
+  });
+
+  const actionsCount = totalActionData?.searchColonyActions?.total ?? 0;
```

Comment:
I think this is causing the 0 flicker @arrenv mentioned.
I guess we can either memo the last value or return a loading flag with the count, displaying a spinner in the widgets?
Otherwise gj on this one, especially with the team -> domain word change, one layer of obfuscation gone :tada: